dockers/docker-nat/restore_nat_entries.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+
+""""
+Description: restore_nat_entries.py -- restoring nat entries table into kernel during system warm reboot.
+    The script is started by supervisord in nat docker when the docker is started.
+    It does not do anything in case neither system nor nat warm restart is enabled.
+    In case nat warm restart enabled only, it sets the stateDB flag so natsyncd can continue
+    the reconciation process.
+    In case system warm reboot is enabled, it will try to restore the nat entries table into kernel
+    , then it sets the stateDB flag for natsyncd to continue the
+    reconciliation process.
+"""
+
+import sys
+import subprocess
+import swsssdk
+from swsscommon import swsscommon
+import logging
+import re
+import os
+
+logging.getLogger("scapy.runtime").setLevel(logging.ERROR)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+logger.addHandler(logging.NullHandler())
+
+def add_nat_conntrack_entry_in_kernel(ipproto, srcip, dstip, srcport, dstport, natsrcip, natdstip, natsrcport, natdstport):
+    # pyroute2 doesn't have support for adding conntrack entries via netlink yet. So, invoking the conntrack utility to add the entries.
+    state = ''
+    if (ipproto == '6'):
+        state = ' --state ESTABLISHED '
+    ctcmd = 'conntrack -I -n ' + natdstip + ':' + natdstport + ' -g ' + natsrcip + ':' + natsrcport + \
+                       ' --protonum ' + ipproto + state + ' --timeout 600 --src ' + srcip + ' --sport ' + srcport + \
```

Comment:
Why there is a hard coded timeout for 600 sec and how it is related to reconciliation timer?

Reply:
@stepanblyschak 600 secs is the default NAT entry timeout. Big enough value to keep the entry from timing out before the application resets the active entry timeouts. And this timeout is not related to the reconciliation timer.

Reply:
Isn't this value configurable through cfg db according to design?

Reply:
@stepanblyschak Yes, this value is configurable through config db, but the config tables will not be available yet (daemons yet to come up) when the restore script is run on bootup. Moreover in case of default values, they will not be stored in CONFIG_DB.
Like mentioned above, this is just the initial timeout value for the restored entries, and it will be reset with the actual configured timeout in the next 30 secs by the application.